docs/src/main/sphinx/object-storage/legacy-s3.md

```diff
@@ -140,12 +145,10 @@ connector property.
 
 The class must implement the
 [AWSCredentialsProvider](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentialsProvider.html)
-interface and provide a two-argument constructor that takes a
-`java.net.URI` and a Hadoop `org.apache.hadoop.conf.Configuration`
-as arguments. A custom credentials provider can be used to provide
-temporary credentials from STS (using `STSSessionCredentialsProvider`),
```

Comment:
trino-hdfs is legacy & to be removed rather sooner than later. if this was broken and didn't work, maybe no need to fix it? Just remove incorrect stuff from the docs?

Reply:
It worked, just was hard to use. This feature is missing in new native fs library, so I'd also like to discuss if we want to add it there. Being able to choose a single credentials provider instead of relying on the (default) chain solves some issues.

How soon are we going to remove this, weeks or months?

Reply:
I'd hope it's sooner than later (so weeks rather than months). 
I am not against having this feature (done correctly) in the new native fs library, if it solves problems. Would it be possible to have an issue describing those problems?

Reply:
See the issue linked in the PR description

Reply:
Thanks. The linked issue is sufficient to me to justify adding the configuration to the new s3 filesystem. @nineinchnick do you plan on working on this?